src/iterator.rs

```diff
@@ -5,34 +5,32 @@ use crate::error::GoResult;
 use crate::gas_meter::gas_meter_t;
 use crate::memory::Buffer;
 
-// this represents something passed in from the caller side of FFI
-#[repr(C)]
-pub struct iterator_t {
-    _private: [u8; 0],
-}
+// iterator is now a simple number (index to pointer on Go side)
+#[allow(non_camel_case_types)]
+pub type iterator_t = i64;
```

Comment:
In my experience using `type` in rust for anything except simplifying generic types (e.g. `Result`s) and defining associated types on traits is a bad pattern that lets you accidentally mix semantically-different types that have the same memory representation.
A more type safe way of representing this is:
```rust
#[allow(non_camel_case_types)]
#[repr(C)]
pub struct iterator_t(i64);
```
Check the bindings.h file though to see what it generates. You may have to add `pub` before the `i64`.
Also, shouldn't this be `u64`?

Reply:
I will make a proper struct again to hold the two counters as I described.

Reply:
oh yeah, ok